src/tests/Common/CoreCLRTestLibrary/Vectors.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace TestLibrary
+{
+    public static class Vectors
+    {
+        public static void VectorToArray<T>(ref T[] to, Vector<T> from)
+        {
+            long tsize = Unsafe.SizeOf<T>();
+            long vsize = Unsafe.SizeOf<Vector<T>>();
+            if (to.Length * tsize != vsize)
+            {
+                throw new ArgumentException("sizeof(to) != sizeof(from)");
+            }
+            Unsafe.WriteUnaligned(ref Unsafe.As<T, byte>(ref to[0]), from);
+        }
+
+        public static void ArrayToVector<T>(ref Vector<T> to, T[] from)
+        {
+            long tsize = Unsafe.SizeOf<T>();
+            long vsize = Unsafe.SizeOf<Vector<T>>();
+            if (from.Length * tsize != vsize)
+            {
+                throw new ArgumentException("sizeof(to) != sizeof(from)");
+            }
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<Vector<T>, byte>(ref to), ref Unsafe.As<T, byte>(ref from[0]), checked((uint)vsize));
+        }
+
+        public static Vector<T> GetRandomVector<T>()
+        {
+            long vsize = Unsafe.SizeOf<Vector<T>>();
+
+            byte[] data = new byte[vsize];
+            for (int i = 0; i < vsize; i++)
+            {
+                data[i] = TestLibrary.Generator.GetByte();
+            }
+
+            Vector<T> vec = new Vector<T>();
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<Vector<T>, byte>(ref vec), ref Unsafe.AsRef<byte>(ref data[0]), checked((uint)vsize));
```

Comment:
Ditto my above comment. `Vector<T>` can be constructed from a `ReadOnlySpan<byte>`:

```suggestion
            return new Vector<T>(data);
```